engine/access/rpc/backend/backend_accounts.go

```diff
@@ -80,6 +80,126 @@ func (b *backendAccounts) GetAccountAtBlockHeight(
+	if err != nil {
+		err := irrecoverable.NewExceptionf("failed to lookup sealed header: %w", err)
+		irrecoverable.Throw(ctx, err)
+		return 0, err
+	}
+
+	sealedBlockID := sealed.ID()
+
+	if b.scriptExecutor != nil {
+		accountBalance, err := b.scriptExecutor.GetAccountBalance(ctx, address, sealed.Height)
+		if err != nil {
+			b.log.Debug().Err(err).Msgf("failed to get account balance at blockID: %v", sealedBlockID)
+			return 0, err
+		}
+
+		return accountBalance, nil
+	}
+
+	account, err := b.getAccountAtBlock(ctx, address, sealedBlockID, sealed.Height)
+	if err != nil {
+		b.log.Debug().Err(err).Msgf("failed to get account balance at blockID: %v", sealedBlockID)
+		return 0, err
+	}
+
+	return account.Balance, nil
+}
+
+// GetAccountBalanceAtBlockHeight returns the account balance at the given block height.
+func (b *backendAccounts) GetAccountBalanceAtBlockHeight(
+	ctx context.Context,
+	address flow.Address,
+	height uint64,
+) (uint64, error) {
+	if b.scriptExecutor != nil {
+		accountBalance, err := b.scriptExecutor.GetAccountBalance(ctx, address, height)
+
+		if err != nil {
+			b.log.Debug().Err(err).Msgf("failed to get account balance at height: %d", height)
+			return 0, err
+		}
+		return accountBalance, nil
+	}
+
+	account, err := b.GetAccountAtBlockHeight(ctx, address, height)
+	if err != nil {
+		return 0, err
+	}
+
+	return account.Balance, nil
+}
+
+// GetAccountKeys returns the account public keys at the latest sealed block.
+// Alias for GetAccountKeysAtLatestBlock
+func (b *backendAccounts) GetAccountKeys(ctx context.Context, address flow.Address) ([]flow.AccountPublicKey, error) {
+	return b.GetAccountKeysAtLatestBlock(ctx, address)
+}
+
+// GetAccountKeysAtLatestBlock returns the account public keys at the latest sealed block.
+func (b *backendAccounts) GetAccountKeysAtLatestBlock(ctx context.Context, address flow.Address) ([]flow.AccountPublicKey, error) {
+	sealed, err := b.state.Sealed().Head()
+	if err != nil {
+		err := irrecoverable.NewExceptionf("failed to lookup sealed header: %w", err)
+		irrecoverable.Throw(ctx, err)
+		return nil, err
+	}
+
+	sealedBlockID := sealed.ID()
+
+	if b.scriptExecutor != nil {
+		accountKeys, err := b.scriptExecutor.GetAccountKeys(ctx, address, sealed.Height)
+		if err != nil {
+			b.log.Debug().Err(err).Msgf("failed to get account keys at blockID: %v", sealedBlockID)
+			return nil, err
+		}
+		return accountKeys, nil
+	}
+
+	account, err := b.getAccountAtBlock(ctx, address, sealedBlockID, sealed.Height)
+	if err != nil {
+		b.log.Debug().Err(err).Msgf("failed to get account keys at blockID: %v", sealedBlockID)
+		return nil, err
+	}
+
+	return account.Keys, nil
+
+}
+
+// GetAccountKeysAtBlockHeight returns the account public keys at the given block height.
+func (b *backendAccounts) GetAccountKeysAtBlockHeight(
+	ctx context.Context,
+	address flow.Address,
+	height uint64,
+) ([]flow.AccountPublicKey, error) {
+	if b.scriptExecutor != nil {
```

Comment:
for both new method types, let's approach this the same way as with get account:

e.g.
```golang
switch b.scriptExecMode {
	case IndexQueryModeExecutionNodesOnly:
		// use method you used when b.scriptExecutor == nil
	case IndexQueryModeLocalOnly:
		// use this method
	case IndexQueryModeFailover:
		// use this method, then failover to the en method on error
```

and we can skip compare mode